server/api/hooks.js

```diff
@@ -1,10 +1,56 @@
 // @flow
 import Router from 'koa-router';
 import httpErrors from 'http-errors';
-import { Document, User } from '../models';
-
+import { Authentication, Document, User } from '../models';
+import * as Slack from '../slack';
 const router = new Router();
 
+router.post('hooks.unfurl', async ctx => {
+  const { challenge, token, event } = ctx.body;
+  if (challenge) return (ctx.body = ctx.body.challenge);
+
+  if (token !== process.env.SLACK_VERIFICATION_TOKEN)
+    throw httpErrors.BadRequest('Invalid token');
+
+  // TODO: Everything from here onwards will get moved to an async job
+  const user = await User.find({ where: { slackId: event.user } });
+  if (!user) return;
+
+  const auth = await Authentication.find({
+    where: { serviceId: 'slack', teamId: user.teamId },
+  });
+  if (!auth) {
+    await Slack.post('chat.unfurl', {
+      token: user.slackAccessToken,
+      channel: event.channel,
+      ts: event.message_ts,
+      user_auth_required: true,
+    });
+    return;
+  }
+
+  // get content for unfurled links
+  let unfurls = {};
+  for (let link of event.links) {
+    const id = link.url.substr(link.url.lastIndexOf('/') + 1);
+    const doc = await Document.findById(id);
+    if (!doc || doc.teamId !== user.teamId) continue;
+
+    unfurls[link.url] = {
+      title: doc.title,
+      text: doc.getSummary(),
+      color: doc.collection.color || undefined,
```

Comment:
Is there a difference between null and undefined here?

Reply:
Not sure, probably not - was just being cautious